### PR TITLE
Let numpy ufuncs operate on HyperSpy BaseSignal instances

### PR DESCRIPTION
**Pull request by francisco-dlp**
This enables operating directly on the HyperSpy's `BaseSignal` instances with numpy ufuncs:

```python

>>> s = hs.signals.Signal1D([0, 1])
>>> s.metadata.General.title = "A"
>>> s
<Signal1D, title: A, dimensions: (|2)>
>>> np.exp(s)
<Signal1D, title: exp(A), dimensions: (|2)>
>>> np.exp(s).data
array([ 1.        ,  2.71828183])
>>> np.power(s, 2)
<Signal1D, title: power(A, 2), dimensions: (|2)>
>>> np.add(s, s)
<Signal1D, title: add(A, A), dimensions: (|2)>
>>> np.add(hs.signals.Signal1D([0, 1]), hs.signals.Signal1D([0, 1]))
<Signal1D, title: add(Untitled Signal 1, Untitled Signal 2), dimensions: (|2)>

```

Notice the the title is automatically updated.

There is a caveat: numpy functions which are not ufuncs can also operate on the HyperSpy signals but I didn't find the way to make them return a HyperSpy signal e.g.:

```python

>>> np.angle(s)
array([ 0.,  0.])

```

Notice that I have also fixed the simple mathematical operations section of the User Guide.

### CODE REVIEW

**Comment by francisco-dlp** [comment]
Notice that the tests are only failing because the EELSdb certificate seems to have expired.

**Comment by francisco-dlp** [comment]
This is ready for review.

File: doc/user_guide/tools.rst [hunk 1]
```diff
@@ -261,13 +277,55 @@ Most of them are just wrapped numpy functions, as an example:
...
+    array([ 1.        ,  2.71828183])
+    >>> np.power(s, 2)
+    <Signal1D, title: power(A, 2), dimensions: (|2)>
+    >>> np.add(s, s)
+    <Signal1D, title: add(A, A), dimensions: (|2)>
+    >>> np.add(hs.signals.Signal1D([0, 1]), hs.signals.Signal1D([0, 1]))
+    <Signal1D, title: add(Untitled Signal 1, Untitled Signal 2), dimensions: (|2)>
+
+
+Notice the the title is automatically updated. When the signal has no title
```
Comment (to266) [inline doc/user_guide/tools.rst]: ...Notice __that__ the title...

**Comment by CodeMonkeyJan** [comment]
Apparently ufuncs can be generated by using the `np.frompyfunc` factory method (see [here](http://docs.scipy.org/doc/numpy/reference/generated/numpy.frompyfunc.html#numpy.frompyfunc)) but I assume this will not really help, because the ufuncs invoke the `__array__` functions in `BaseSignal` and non-ufunc functions will not, thus robbing us of the hook to replace them with ufunc versions.

Overall this is a great addition as it allows very intuitive usage of numpy with the signals :-)!

**Comment by francisco-dlp** [comment]
The typo should be fixed now.

**Pull request merged**